WordPress/Classes/ViewRelated/Pages/PageListViewController.swift

```diff
@@ -580,4 +538,72 @@ class PageListViewController : AbstractPostListViewController, UIViewControllerR
             restorePost(apost)
         }
     }
+
+    // MARK: - Refreshing noResultsView
+
+    func handleRefreshNoResultsView(noResultsView: WPNoResultsView) {
+        noResultsView.titleText = noResultsTitle()
+        noResultsView.messageText = noResultsMessage()
+        noResultsView.accessoryView = noResultsAccessoryView()
+        noResultsView.buttonTitle = noResultsButtonTitle()
+    }
+
+    // MARK: - NoResultsView Customizer helpers
+
+    private func noResultsAccessoryView() -> UIView {
+        if syncHelper.isSyncing {
+            animatedBox.animateAfterDelay(0.1)
+            return animatedBox
+        }
+
+        return UIImageView(image: UIImage(named: "illustration-posts"))
+    }
+
+    private func noResultsButtonTitle() -> String {
+        if syncHelper.isSyncing == true || isSearching() {
+            return ""
+        }
+
+        let filterType = currentPostListFilter().filterType
+
+        switch filterType {
+        case .Trashed:
+            return ""
+        default:
+            return NSLocalizedString("Start a Page", comment: "Button title, encourages users to create their first page on their blog.")
+        }
+    }
+
+    private func noResultsTitle() -> String {
+        if syncHelper.isSyncing == true {
+            return NSLocalizedString("Fetching pages...", comment: "A brief prompt shown when the reader is empty, letting the user know the app is currently fetching new pages.")
+        }
+
+        let filter = currentPostListFilter()
+        let titles = noResultsTitles()
+        let title = titles[filter.filterType]
+        return title ?? ""
+    }
+
+    private func noResultsMessage() -> String {
+        if syncHelper.isSyncing == true || isSearching() {
+            return ""
+        }
+
+        let filterType = currentPostListFilter().filterType
+        var message : String
+
+        switch filterType {
+        case .Draft:
+            message = NSLocalizedString("Would you like to create one?", comment: "Displayed when the user views drafts in the pages list and there are no pages")
+        case .Scheduled:
+            message = NSLocalizedString("Would you like to schedule a draft to publish?", comment: "Displayed when the user views scheduled pages in the oages list and there are no pages")
+        case .Trashed:
+            message = NSLocalizedString("Everything you write is solid gold.", comment: "Displayed when the user views trashed pages in the pages list and there are no pages")
+        default:
+            message = NSLocalizedString("Would you like to publish your first page?", comment: "Displayed when the user views published pages in the pages list and there are no pages")
```

Comment:
Nipticky: how about if we just `return` right there, in each case, instead of assigning to `message`?